msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/GetAccessTokenTests.kt

```diff
@@ -0,0 +1,371 @@
+        val tokenWithEmployeeScope = authResult1.accessToken
+
+        // Reset
+        wasRenewAccessTokenInvoked = false
+
+        // Token should be retrieved from API, as the scope belongs to a different API resource
+        val getAccessTokenResult2 = accountState.getAccessToken(
+            forceRefresh = false,
+            scopes = listOf(CUSTOMERS_READ_ALL_SCOPE)
+        )
+        assertState(getAccessTokenResult2, GetAccessTokenResult.Complete::class.java)
+        Assert.assertTrue(wasRenewAccessTokenInvoked)
+        val authResult2 = (getAccessTokenResult2 as GetAccessTokenResult.Complete).resultValue
+        Assert.assertTrue(authResult2.scope.contains(CUSTOMERS_READ_ALL_SCOPE))
+        val tokenWithCustomerScope = authResult2.accessToken
+        Assert.assertNotEquals(tokenWithEmployeeScope, tokenWithCustomerScope)
+    }
+
+    @Test
+    fun testGetAccessTokenWith2CustomApiResources() = runTest {
+        val username = NativeAuthCredentialHelper.nativeAuthSignInUsername
+        val password = getSafePassword()
+        val result = application.signIn(
+            username = username,
+            password = password.toCharArray()
+        )
+        assertState(result, SignInResult.Complete::class.java)
+        val accountState = (result as SignInResult.Complete).resultValue
+
+        // Var to keep track of whether BaseController.renewAccessToken() was called. This method calls the API to refresh the access token, for example if it's expired or not available in cache.
+        var wasRenewAccessTokenInvoked = false
+        ShadowBaseController.setOnRenewAccessTokenInvokedCallback { wasRenewAccessTokenInvoked = true }
+
+        // Token should be retrieved from API
+        val getAccessTokenResult1 = accountState.getAccessToken(scopes = listOf(EMPLOYEE_WRITE_ALL_SCOPE))
+        assertState(getAccessTokenResult1, GetAccessTokenResult.Complete::class.java)
+        Assert.assertTrue(wasRenewAccessTokenInvoked)
+        val authResult1 = (getAccessTokenResult1 as GetAccessTokenResult.Complete).resultValue
+        Assert.assertTrue(authResult1.scope.contains(EMPLOYEE_WRITE_ALL_SCOPE))
+        val tokenWithEmployeeScope = authResult1.accessToken
+
+        // Reset
+        wasRenewAccessTokenInvoked = false
+
+        // Token should be retrieved from API, as the scope belongs to a different API resource
+        val getAccessTokenResult2 = accountState.getAccessToken(forceRefresh = false, scopes = listOf(CUSTOMERS_READ_ALL_SCOPE))
+        assertState(getAccessTokenResult2, GetAccessTokenResult.Complete::class.java)
+        Assert.assertTrue(wasRenewAccessTokenInvoked)
+        val authResult2 = (getAccessTokenResult2 as GetAccessTokenResult.Complete).resultValue
+        Assert.assertTrue(authResult2.scope.contains(CUSTOMERS_READ_ALL_SCOPE))
+        val tokenWithCustomerScope = authResult2.accessToken
+        Assert.assertNotEquals(tokenWithEmployeeScope, tokenWithCustomerScope)
+    }
```

Comment:
I can't see a difference between this test and the previous one?

Reply:
The naming could be better, but the previous test signs in with custom scope 1, then retrieves token for that same scope from cache, and then retrieves a token for custom scope 2 from API. This test signs in with no scopes, then retrieves token for scope 1 from API, and then token for scope 2 from API. Let me add some comments to the tests to explain the steps, for readability.